packages/fetchai/skills/generic_buyer/behaviours.py

```diff
@@ -92,3 +95,64 @@ def teardown(self) -> None:
         :return: None
         """
         pass
+
+
+class GenericTransactionBehaviour(TickerBehaviour):
+    """A behaviour to sequentially submit transactions to the blockchain."""
+
+    def __init__(self, **kwargs):
+        """Initialize the transaction behaviour."""
+        tx_interval = cast(float, kwargs.pop("tx_interval", DEFAULT_TX_INTERVAL))
+        self.max_processing = cast(
+            float, kwargs.pop("max_processing", DEFAULT_MAX_PROCESSING)
+        )
+        self.processing_time = 0.0
+        self.waiting: List[Tuple[LedgerApiDialogue, LedgerApiMessage]] = []
+        self.processing: Optional[LedgerApiDialogue] = None
+        super().__init__(tick_interval=tx_interval, **kwargs)
+
+    def setup(self) -> None:
+        """Setup behaviour."""
+        pass
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+        if self.processing is not None and self.processing_time <= self.max_processing:
+            # already processing
+            self.processing_time += self.tick_interval
```

Comment:
is this safe? For our purposes I'd say it's good enough. A more resilient approach would be to take the current time and the last time `act` was called; the reason is that `tick_interval` might not be exactly the time elapsed between two acts, especially in case the AEA runtime is overloaded. 

Another point in favour of the current code is that `processing_time` would be an underestimation of the actual time elapsed, which means we would wait more than we did (which is safe to do in this case, but less responsive).

Reply:
I think it is safe in the sense that it will eventually terminate; it won't necessarily be exactly after max_processing seconds, but that;'s ok.